Sources/Lifecycle/Lifecycle.swift

```diff
@@ -82,15 +84,101 @@ public struct LifecycleHandler {
         return LifecycleHandler(nil)
     }
 
-    internal func run(_ callback: @escaping (Error?) -> Void) {
-        let body = self.body ?? { callback in
+    internal func run(_ completionHandler: @escaping (Error?) -> Void) {
+        let body = self.underlying ?? { callback in
             callback(nil)
         }
-        body(callback)
+        body(completionHandler)
     }
 
     internal var noop: Bool {
-        return self.body == nil
+        return self.underlying == nil
+    }
+}
+
+// MARK: - Stateful Lifecycle Handlers
+
+/// LifecycleHandler for starting stateful tasks. The state can then be fed into a LifecycleShutdownHandler
+public struct LifecycleStartHandler<State> {
+    public typealias Handler = (@escaping (Result<State, Error>) -> Void) -> Void
```

Comment:
I would not use `typealias` for functions, especially not in public API. The reason is that in Xcode you don't get useful completions. Ie the user won't know what type of function they have to implement. They'll have to Cmd-click `Handler`, then copy and paste the type. It's harder to write (for you) without the typealias but it's much easier for users to work with it.

Reply:
good idea. addressed